### PR TITLE
Patch Style and updates version 0.5

### PR DESCRIPTION
**Pull request by naomijub**
Hi @tasgon, I updated the Cargo.toml to version 0.5 and did some minor coding style changes that make your code more expressive.

Also:
- Added const to a couple of functions that were const.
- Added `_else` clause to some of your `map/or` because they were allocating memory and it was not necessary to do it eagerly.
- Ran `cargo update` to update your `Cargo.lock`

### CODE REVIEW

File: examples/fonts.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ pub fn main() {
 
 fn ui_system(mut ctx: IcedContext<UiMessage>) {
     ctx.display(column!(
-        text(format!("I am the default font")).font(font::Font::DEFAULT),
-        text(format!("I am another font"))
```
Comment (tasgon) [inline examples/fonts.rs:34]: Are there any technical reasons to prefer `to_string` over `format!`?

File: src/utils.rs [hunk 2]
```diff
@@ -3,7 +3,7 @@ use crate::IcedContext;
 use bevy_math::Vec2;
 use bevy_window::Window;
 
-pub(crate) fn process_cursor_position(
```
Comment (tasgon) [inline src/utils.rs:6]: Why remove the `(crate)` modifiers?

File: src/lib.rs [hunk 3]
```diff
@@ -247,14 +247,12 @@ impl<'w, 's, M: bevy_ecs::event::Event> IcedContext<'w, 's, M> {
...
-                Some(position) => {
+            window.cursor_position().map_or_else(
+                || utils::process_touch_input(self).map_or(Cursor::Unavailable, Cursor::Available),
+                |position| {
                     Cursor::Available(utils::process_cursor_position(position, bounds, window))
-                }
-                None => utils::process_touch_input(self)
-                    .map(Cursor::Available)
-                    .unwrap_or(Cursor::Unavailable),
-            }
```
Comment (tasgon) [inline src/lib.rs:257]: Is there a performance advantage to `map_or_else`? As far as I can tell, [`map_or_else` compiles down to a `match` statement](https://doc.rust-lang.org/src/core/option.rs.html#1171-1174), and stylistically speaking, I prefer the `match` version.

File: examples/fonts.rs (hunk 1 above)
Comment (naomijub) [reply examples/fonts.rs:34]: By using format you are overriding the specialized to_string function. A general good practice in this case would be to use to_string unless you need some extra formatting

Comment (naomijub) [reply examples/fonts.rs:34]: By using format you are overriding the specialized to_string function. A general good practice in this case would be to use to_string unless you need some extra formatting

File: src/utils.rs (hunk 2 above)
Comment (naomijub) [reply src/utils.rs:6]: The module is already private, it is a bit misleading having pub(crate) inside private module.

File: src/lib.rs (hunk 3 above)
Comment (naomijub) [reply src/lib.rs:257]: It does, but it is still "protected" inside the lazy evaluation of the closure. And that function does quite a lot of work. But I agree, match is cleaner, when you have a value that you already own or is already allocated

**Comment by naomijub** [comment]
@tasgon Do you want me to chance the comment you made or did my responses clarify?

File: src/lib.rs (hunk 3 above)
Comment (tasgon) [reply src/lib.rs:257]: So, why don't we leave it as the original `match` version?

Comment (naomijub) [reply src/lib.rs:257]: No problem, will do the change tomorrow

Comment (naomijub) [reply src/lib.rs]: af7ea2f @tasgon

**Comment by naomijub** [comment]
@tasgon ?

**Comment by tasgon** [comment]
Sorry --  looks good. I'll double check it tomorrow, and if I don't see anything else I'll merge and publish.

**Pull request merged**